server/src/main/java/com/hedera/block/server/persistence/StreamPersistenceHandlerImpl.java

```diff
@@ -0,0 +1,158 @@
+        this.serviceStatus = serviceStatus;
+    }
+
+    /**
+     * The onEvent method is invoked by the Disruptor when a new SubscribeStreamResponse is
+     * available. The method processes the response and persists the block item to the file system.
+     *
+     * @param event the ObjectEvent containing the SubscribeStreamResponse
+     * @param l the sequence number of the event
+     * @param b true if the event is the last in the sequence
+     */
+    @Override
+    public void onEvent(ObjectEvent<SubscribeStreamResponse> event, long l, boolean b) {
+        try {
+            if (serviceStatus.isRunning()) {
+
+                final SubscribeStreamResponse subscribeStreamResponse = event.get();
+                final OneOf<SubscribeStreamResponse.ResponseOneOfType> oneOfTypeOneOf =
+                        subscribeStreamResponse.response();
+                switch (oneOfTypeOneOf.kind()) {
+                    case BLOCK_ITEM -> {
+                        final BlockItem blockItem = subscribeStreamResponse.blockItem();
+                        if (blockItem == null) {
+                            final String message =
+                                    PROTOCOL_VIOLATION_MESSAGE.formatted(
+                                            "SubscribeStreamResponse",
+                                            "BLOCK_ITEM",
+                                            "block_item",
+                                            subscribeStreamResponse);
+                            LOGGER.log(ERROR, message);
+                            metricsService.get(StreamPersistenceHandlerError).increment();
+                            throw new BlockStreamProtocolException(message);
+                        } else {
+                            // Persist the BlockItem
+                            Optional<BlockItem> result = blockWriter.write(blockItem);
+                            if (result.isPresent()) {
+                                // Publish the block item back upstream to the notifier
+                                // to send responses to producers.
+                                notifier.publish(blockItem);
+                            }
+                        }
+                    }
+                    case STATUS -> LOGGER.log(
+                            DEBUG, "Unexpected received a status message rather than a block item");
+                    default -> {
+                        final String message = "Unknown response type: " + oneOfTypeOneOf.kind();
+                        LOGGER.log(ERROR, message);
+                        metricsService.get(StreamPersistenceHandlerError).increment();
+                        throw new BlockStreamProtocolException(message);
+                    }
+                }
+            } else {
+                LOGGER.log(
+                        ERROR, "Service is not running. Block item will not be processed further.");
```

Comment:
Same here

Reply:
See my previous comment, the behavior was intended to be the same. We log a simple message indicating that this component was no longer processing but I think throwing an exception just adds confusion to the logs. Also, this method is called by the RingBuffer when it receives a new block item. It's really a one-way street designed for performance. It doesn't handle checked exceptions.

Reply:
Gotcha, thank you for the answer 👍